ocfweb/api/session_tracking.py

```diff
@@ -0,0 +1,111 @@
+import json
+from functools import partial
+from ipaddress import ip_address
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.http import HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from ocflib.infra.ldap import ldap_ocf
+from ocflib.infra.ldap import OCF_LDAP_HOSTS
+from ocflib.infra.net import is_ocf_ip
+from ocflib.lab.stats import get_connection
+
+from ocfweb.caching import cache
+
+get_connection = partial(
+    get_connection,
+    user=settings.OCFSTATS_USER,
+    password=settings.OCFSTATS_PASSWORD,
+    db=settings.OCFSTATS_DB,
+)
+
+
+@require_POST
+@csrf_exempt
+def log_session(request):
+    """Primary API endpoint for session tracking.
+
+    Desktops have a cronjob that calls this endpoint, replacing
+    the functionality that used to be in ocf/labstats.
+    """
+
+    if not is_ocf_ip(ip_address(request.META['REMOTE_ADDR'])):
+        return HttpResponse('Not Authorized', status=401)
+
+    try:
+        body = json.loads(request.body.decode('utf-8'))
+
+        host = _get_desktops().get(request.META['REMOTE_ADDR'], None)
+        user = body.get('user', None)
+
+        if user is '' or user is None:
+            raise ValueError('Invalid user "{}"'.format(user))
+
+        if host is None:
+            raise ValueError('Invalid host "{}"'.format(host))
+
+        if _session_exists(host, user):
+            _refresh_session(host, user)
+        else:
+            _new_session(host, user)
+
+        return HttpResponse(status=200)
+
+    except Exception as e:
```

Comment:
please be less greedy here. what exceptions are we guarding against and in what code?

as written, this is likely to mask real errors.

Reply:
Would `ValueError` be appropriate? The two main things I was thinking of were the ones I raise and `JSONDecodeError`, which subclasses `ValueError`.

Reply:
instead of raising ValueError, can you just return a 400 bad request there?

there are lots of things that raise valueerrors

Reply:
(my main concern is that you'll catch a legitimate exception and it'll return a bad request, which nobody will ever notice, and then we'll have a real error for a long time with nobody noticing or fixing it)

Reply:
that's certainly true, this was just broken for several hours and I didn't notice because the errors were being sent to the client and from there promptly to /dev/null.